### PR TITLE
Fix for the template map concurrency issue

### PR DESCRIPTION
**Pull request by runner-mei**
fix https://github.com/revel/revel/issues/1188

### CODE REVIEW

File: template.go [hunk 1]
```diff
@@ -253,7 +289,7 @@ func (loader *TemplateLoader) loadIntoEngine(engine TemplateEngine, baseTemplate
 		return
 	}
 	if err = engine.ParseAndAdd(baseTemplate); err == nil {
-		loader.TemplatePaths[baseTemplate.TemplateName] = baseTemplate.FilePath
+		runtime.TemplatePaths[baseTemplate.TemplateName] = baseTemplate.FilePath
```
Comment (notzippy) [inline template.go]: add in 
```
if tmpl := engine.Lookup(langName);tmpl!=nil {
    runtime.TemplateMap[baseTemplate.TemplateName]= tmpl 
}
```

File: template.go [hunk 2]
```diff
@@ -253,7 +289,7 @@ func (loader *TemplateLoader) loadIntoEngine(engine TemplateEngine, baseTemplate
 		return
 	}
```
Comment (notzippy) [inline template.go]: Add in check to see if template map contains the template already, if so do not add it
```
 	if loadedTemplate, found := runtime.TemplateMap[baseTemplate.TemplateName]; found!=false {
 		// Duplicate template found in map
 		TRACE.Println("template already exists in map: ", baseTemplate.TemplateName, " in engine ", engine.Name(), "\r\n\told file:",
                loadedTemplate.Location(), "\r\n\tnew file:", baseTemplate.FilePath)
 		return
 	}
```

Comment (runner-mei) [inline template.go]: I add it!

File: template.go (hunk 1 above)
Comment (runner-mei) [inline template.go]: I add it!

**Comment by runner-mei** [comment]
I rebase it with revel/develop branch.

**Pull request merged**